app/models/concerns/twitch_user.rb

```diff
@@ -4,50 +4,41 @@
 module TwitchUser
   extend ActiveSupport::Concern
 
-  included do
-    def sync_twitch_follows!
-      success = false
-      ActiveRecord::Base.transaction do
-        current_followed_users = User.where(twitch_id: Twitch::Follows.followed_ids(twitch_id))
-        old_followed_users = twitch_followed_users
-
-        TwitchUserFollow.import((current_followed_users - old_followed_users).map do |u|
-          TwitchUserFollow.new(from_user: self, to_user: u)
-        end)
-
-        (old_followed_users - current_followed_users).each do |u|
-          TwitchUserFollow.find_by(from_user: self, to_user: u).destroy
-        end
-
-        success = true
-      end
-      update(twitch_user_follows_checked_at: nil) unless success
-    end
-
-    def twitch_sync!
-      body = Twitch::User.get(twitch_id)
+  def sync_twitch_follows!
+    ActiveRecord::Base.transaction do
+      TwitchUserFollow.where(from_user: self, to_user: (twitch_followed_users - current_followed_users)).delete_all
 
-      basic_info = JSON.parse(body)
-
-      twitch_id = basic_info['_id']
-      name = basic_info['name']
-      avatar = basic_info['logo']
-      display_name = basic_info['display_name']
+      TwitchUserFollow.import!((current_followed_users - twitch_followed_users).map do |u|
+        TwitchUserFollow.new(from_user: self, to_user: u)
+      end)
+    end
+  rescue ActiveRecord::RecordInvalid, ActiveRecord::StatementInvalid => e
+    update(twitch_user_follows_checked_at: nil)
+    raise e
+  end
 
-      avatar = 'https://static-cdn.jtvnw.net/jtv_user_pictures/xarth/404_user_150x150.png' if avatar.nil?
+  def twitch_sync!
+    body = JSON.parse(Twitch::User.get(twitch_id))
+
+    update(
+      twitch_id:           body['_id'],
+      name:                body['name'].downcase,
+      avatar:              URI.parse(body['logo'] || default_avatar).tap { |uri| uri.scheme = 'https' }.to_s,
+      twitch_display_name: body['display_name']
+    )
+  rescue RestClient::ResourceNotFound
+    nil
+  end
 
-      uri = URI.parse(avatar).tap do |uri_tap|
-        uri_tap.scheme = 'https'
-      end
+  private
 
-      update(
-        twitch_id: twitch_id,
-        name: name.downcase,
-        avatar: uri.to_s,
-        twitch_display_name: display_name
-      )
-    rescue RestClient::ResourceNotFound
-      nil
+  def current_followed_users
+    Rails.cache.fetch("#{id}/follows") do
```

Comment:
If we don't pass the ABC score in a lot of places, it might be worth just doing inline ignores on specific methods that way it still keeps other code to a higher standard.

I would agree that caching this is not great since AR caches queries already, and depending on how the cache is set up in production this might be slower than what AR does already.